fe1-web/src/features/rollCall/screens/ViewSingleRollCall.tsx

```diff
@@ -48,337 +38,33 @@ const ViewSingleRollCall = () => {
     throw new Error(`Could not find a roll call with id ${rollCallId}`);
   }
 
-  const navigation = useNavigation<NavigationProps['navigation']>();
-
   const laoId = RollCallHooks.useAssertCurrentLaoId();
-  const generateToken = RollCallHooks.useGenerateToken();
-  const hasSeed = RollCallHooks.useHasSeed();
-
-  if (!laoId) {
-    throw new Error('no LAO is currently active');
-  }
-  const [popToken, setPopToken] = useState('');
-  const [hasWalletBeenInitialized, setHasWalletBeenInitialized] = useState(hasSeed());
-
-  // re-check if wallet has been initialized after focus events
-  useEffect(() => {
-    // Return the function to unsubscribe from the event so it gets removed on unmount
-    return navigation.addListener('focus', () => {
-      setHasWalletBeenInitialized(hasSeed());
-    });
-  }, [navigation, hasSeed]);
 
-  useEffect(() => {
-    if (!hasWalletBeenInitialized || !laoId || !rollCall?.id) {
-      return;
-    }
-
-    // Here we get the pop-token to display in the QR code
-    generateToken(laoId, rollCall.id)
-      .then((token) => setPopToken(token.publicKey.valueOf()))
-      .catch((err) => console.error(`Could not generate token: ${err}`));
-  }, [hasWalletBeenInitialized, generateToken, laoId, rollCall]);
-
-  if (!rollCall) {
-    console.debug('Error in Roll Call display: Roll Call doesnt exist in store');
-    return null;
+  switch (rollCall.status) {
+    case RollCallStatus.CREATED:
+      return <RollCallCreated rollCall={rollCall} laoId={laoId} isOrganizer={isOrganizer} />;
+    case RollCallStatus.REOPENED:
+    case RollCallStatus.OPENED:
+      return (
+        <RollCallOpen
+          rollCall={rollCall}
+          laoId={laoId}
+          isOrganizer={isOrganizer}
+          scannedPopTokens={attendeePopTokens}
+        />
+      );
+    case RollCallStatus.CLOSED:
+      return <RollCallClosed rollCall={rollCall} laoId={laoId} isOrganizer={isOrganizer} />;
+    default:
+      console.warn('Roll Call Status was undefined in EventRollCall');
```

Comment:
not necessarily undefined, it could also be (in the future) a new value that's unsupported here.

Reply:
true!